File: binary_search.py
import math

def binary_search(lst, item):
    low = 0
    high = len(lst) - 1
    while low <= high:
        mid = (low + high)
        guess = lst[int(round(mid))]
        if guess == item:
            return mid
        if guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None

File: binary_search.js
function binarySearch(list, target) {
  let low = 0
  let high = list.length - 1
  while (low <= high) {
    let mid = Math.round((low + high) / 2)
    let guess = list[mid]
    if (guess == target) {
      return mid
    }
    if (guess > target) {
      high = mid - 1
    } else low = mid + 1
  }
  return -1
}
